pyrenew_hew/utils.py

```diff
@@ -17,21 +20,250 @@
+
+    Parameters
+    ----------
+    hew_letters
+        The relevant HEW letters, or 'null'.
+
+    flag_prefix
+        Prefix for the names of the boolean flags in the
+        output dictionary, which will be
+        ``{flag_prefix}_hospital_admissions``,
+        ``{flag_prefix}_ed_visits``, and
+        ``{flag_prefix}_wastewater``. Default ``"fit"``,
+        since {H,E,W} models are named according to the
+        datastreams used in fitting.
+
+    Returns
+    -------
+    dict[str, bool]
+        Dictionary of boolean values with keys named
+        ``{flag_prefix}_hospital_admissions``,
+        ``{flag_prefix)_ed_visits``, and
+        ``{flag_prefix}_wastewater`` (with the given
+        ``flag_prefix``).
+
+    Raises
+    ------
+    ValueError
+        if input does is neither ``'null'`` nor a
+        combination of the letters
+        ``'h'``, ``'e'``, and ``'w'``.
+    """
+    validate_hew_letters(hew_letters)
+    return {
+        f"{flag_prefix}_hospital_admissions": "h" in hew_letters,
+        f"{flag_prefix}_ed_visits": "e" in hew_letters,
+        f"{flag_prefix}_wastewater": "w" in hew_letters,
+    }
+
+
+def pyrenew_model_name_from_flags(
+    fit_ed_visits: bool = False,
+    fit_hospital_admissions: bool = False,
+    fit_wastewater: bool = False,
+) -> str:
+    """
+    Get a "pyrenew_{h,e,w}" model name
+    string from a set of flags indicating which
+    of the datastreams, if any, were used in fitting.
+    If none of them were, call the model "pyrenew_null".
+
+    Parameters
+    ----------
+    fit_ed_visits
+        ED visit data used in fitting?
+
+    fit_hospital_admissions
+        Hospital admissions data used in fitting?
+
+    fit_wastewater
+        Wastewater data used in fitting?
+
+    Returns
+    -------
+    str
+        The model name.
+    """
+    hew_letters = hew_letters_from_flags(
+        fit_ed_visits=fit_ed_visits,
+        fit_hospital_admissions=fit_hospital_admissions,
+        fit_wastewater=fit_wastewater,
+    )
+    return f"pyrenew_{hew_letters}"
+
+
+def flags_from_pyrenew_model_name(model_name: str) -> dict[str, bool]:
+    """
+    Get a set of boolean flags indicating which
+    datastreams, if any, were used in fitting
+    from a "pyrenew_{h,e,w}" model name.
+
+    Parameters
+    ----------
+    model_name
+        The model name.
+
+    Returns
+    -------
+    dict[str, bool]
+       Dictionary with boolean entries named
+       ``fit_hospital_admissions``, ``fit_ed_visits``,
+       and ``fit_wastewater``.
+
+    Raises
+    ------
+    ValueError
+        If the input is not a valid pyrenew_{h, e, w} model name.
+
+    """
+    if not model_name.startswith("pyrenew_"):
+        raise ValueError(
+            "Expected a model_name beginning with "
+            f"'pyrenew_'. Got {model_name}."
+        )
+    hew_letters = model_name.removeprefix("pyrenew_")
+    return flags_from_hew_letters(hew_letters, flag_prefix="fit")
+
+
+def build_pyrenew_model(
```

Comment:
Function naming suggestion
```suggestion
def build_pyrenew_hew_model(
```

Reply:
[`4beecdd` (#578)](https://github.com/CDCgov/pyrenew-hew/pull/578/commits/4beecdd8e700637710e186b8887e698a66307804)